### PR TITLE
Add body generation for ETH JSON RPC calls

### PR DESCRIPTION
**Pull request by bbondy**
This follows the spec on https://eth.wiki/json-rpc/API
I tried running all of these on Infura as well to make sure they are all well formed.
I commented on the calls which aren't available on Infura.  Some are available only on local nodes.
Note that these won't be used by your proxy object, it just gives the browser the ability to speak to an ethereum client.

To review this only review these 2 commits:
- Add body generation for ETH JSON RPC calls
- Add tests for ETH JSON RPC requests

The other ones are from a previous PR that this is based on that will merge soon.

Resolves https://github.com/brave/brave-browser/issues/14282

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/eth_requests_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,331 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (SergeyZhukovsky) [inline components/brave_wallet/eth_requests_unittest.cc:1]: 2021?

Comment (bbondy) [reply components/brave_wallet/eth_requests_unittest.cc]: Fixed and pushed.

**Comment by mkarolin** [comment]
Seems like `patches/chrome-VERSION.patch`, `package.json`, and `package-lock.json` aren't meant to be in this PR?

**Comment by bbondy** [comment]
> Seems like `patches/chrome-VERSION.patch`, `package.json`, and `package-lock.json` aren't meant to be in this PR?

Rebased to remove

**Comment by SergeyZhukovsky** [comment]
post-init failed due to audit, not related to the PR changes

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**